In [2]:
import sklearn

In [3]:
from sklearn.datasets import load_breast_cancer

In [34]:
from sklearn.model_selection import train_test_split

In [35]:
from sklearn.naive_bayes import GaussianNB

In [ ]:
from sklearn.metrics import accuracy_score

In [ ]:
#Load dataset

In [4]:
data = load_breast_cancer()

In [ ]:
#Organize our data

The "data" variable represents a Python object that works like a
dictionary. The important dictionary keys to consider are the
classification label names (target_names), the actual labels (target),
the attribute/feature names (feature_names), and the attributes
(data).
Attributes are a critical part of any classifier. Attributes capture
important characteristics about the nature of the data. Given the label we
are trying to predict (malignant versus benign tumor), possible useful
attributes include the size, radius, and texture of the tumor.

In [9]:
label_names = data['target_names']

In [13]:
labels = data['target']

In [10]:
feature_names  = data['feature_names']

In [11]:
features = data['data']

In [ ]:
#Look at our data

In [12]:
print(label_names)

['malignant' 'benign']


In [14]:
print(labels[0])

0


In [15]:
print(feature_names[0])

mean radius


In [16]:
print(features[0])

[1.799e+01 1.038e+01 1.228e+02 1.001e+03 1.184e-01 2.776e-01 3.001e-01
 1.471e-01 2.419e-01 7.871e-02 1.095e+00 9.053e-01 8.589e+00 1.534e+02
 6.399e-03 4.904e-02 5.373e-02 1.587e-02 3.003e-02 6.193e-03 2.538e+01
 1.733e+01 1.846e+02 2.019e+03 1.622e-01 6.656e-01 7.119e-01 2.654e-01
 4.601e-01 1.189e-01]


To evaluate how well a classifier is performing, you should always test
the model on unseen data. Therefore, before building a model, split your
data into two parts: a training set and a test set.
You use the training set to train and evaluate the model during the
development stage. You then use the trained model to make predictions
on the unseen test set. This approach gives you a sense of the model’s
performance and robustness.

In [18]:
#Split our data

In [19]:
train,test,train_labels,test_labels = train_test_split(features,labels,test_size=0.33,random_state=42)

In [21]:
#Initialize our classifier

In [22]:
gnb = GaussianNB()

In [23]:
#Train our classifier

In [24]:
model = gnb.fit(train,train_labels)

Use the predict() function with the test set and print the results

In [26]:
#Make predictions

In [27]:
preds = gnb.predict(test)

In [28]:
print(preds)

[1 0 0 1 1 0 0 0 1 1 1 0 1 0 1 0 1 1 1 0 1 1 0 1 1 1 1 1 1 0 1 1 1 1 1 1 0
 1 0 1 1 0 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1 0 0 1 1 0 0 1 1 1 0 0 1 1 0 0 1 0
 1 1 1 1 1 1 0 1 1 0 0 0 0 0 1 1 1 1 1 1 1 1 0 0 1 0 0 1 0 0 1 1 1 0 1 1 0
 1 1 0 0 0 1 1 1 0 0 1 1 0 1 0 0 1 1 0 0 0 1 1 1 0 1 1 0 0 1 0 1 1 0 1 0 0
 1 1 1 1 1 1 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1 0 0 1 1 0 1 1 0 1 1 1 1 1 1 0 0
 0 1 1]


In [29]:
#We will use the sklearn function accuracy_score() to determine the accuracy of our machine learning classifier.

In [31]:
#Evaluate accuracy

In [33]:
print(accuracy_score(test_labels,preds))

0.9414893617021277


The NB classifier is 94.15% accurate. This means that 94.15 percent of the time the classifier is able to make the
correct prediction as to whether or not the tumor is malignant or benign.